### ISSUE TITLE
After conda update, ipython now issuing warning at startup

### ISSUE THREAD

**Issue by sizzzzlerz** [ipython/ipython#10851]
So, after doing a "conda update --all",  a whole slew of packages were installed or upgraded. Now, when I issue a "ipython --pylab" command, the following message is displayed:

```
C:\Users\larry\Documents\projects\bhtree>ipython --pylab
C:\Anaconda2\lib\site-packages\win_unicode_console\__init__.py:31: RuntimeWarning: sys.stdin.encoding == 'cp437', whereas sys.stdout.encoding == 'utf-8', readline hook consumer may assume they are the same
  readline_hook.enable(use_pyreadline=use_pyreadline)
Using matplotlib backend: Qt5Agg
```
ipython 5.4.1 is loaded and I'm running python 2.7.13. Also, just entering ipython generates the warning. I'm on Windows 10. Prior to the upgrade, I never saw this message.

**Comment by takluyver** [comment]
Have you noticed any problems? It might be harmless. cc @Drekin

**Comment by Drekin** [comment]
IIRC some places in pyreadline or even in Python input() implementation silently assume that the two encodings are the same (in the interactive case). Why are the encodings different in this case?

**Comment by takluyver** [comment]
It's probably related to some stuff we do on Py 2 to handle stuff printed in the wrong encoding (#9768):

https://github.com/ipython/ipython/blob/5.4.1/IPython/terminal/interactiveshell.py#L384-L399

**Comment by Drekin** [comment]
But that is just wrapping, it doesn't change the `encoding` attribute. So the question is why “sys.stdin.encoding == 'cp437', whereas sys.stdout.encoding == 'utf-8'” holds in OP's setup.

**Comment by takluyver** [comment]
But what it's wrapping is `stdout_text_transcoded`, which is set up to have utf-8 encoding in WUC:

```python
stdout_text_transcoded = TextTranscodingWrapper(stdout_text, encoding="utf-8")
```

**Comment by Drekin** [comment]
Sure, but why is `sys.stdin.encoding` still cp437?

**Comment by takluyver** [comment]
IPython doesn't do anything special with stdin, so that's getting WUC's default treatment, whereas we use the transcoding streams for stdout and stderr.

**Comment by Drekin** [comment]
Yes, but WUC's default treatment is to use transcoding for Python 2 stdin. Every WUC's default variant has the same encoding attribute of all sys.std* objects.

**Comment by takluyver** [comment]
OK, in that case I have no idea! :wink:

**Comment by sizzzzlerz** [comment]
> Have you noticed any problems?
No, my program and plot appear as normal. However, the program does not input text from the user. It does write to the console and there is text in the plot but they both appear normal.
I also updated the anaconda environment on my Mac to the same version and I don't see this message, if that's worth anything.

**Comment by takluyver** [comment]
The code that's issuing the warning is only run on Windows, so I wouldn't expect to see that on a Mac.
